aggregator/aggregator.go

```diff
@@ -267,34 +279,139 @@ func (a *Aggregator) sendFinalProof() {
 
 			log.Infof("Final proof inputs: NewLocalExitRoot [%#x], NewStateRoot [%#x]", inputs.NewLocalExitRoot, inputs.NewStateRoot)
 
-			// add batch verification to be monitored
-			sender := common.HexToAddress(a.cfg.SenderAddress)
-			to, data, err := a.Ethman.BuildTrustedVerifyBatchesTxData(proof.BatchNumber-1, proof.BatchNumberFinal, &inputs, sender)
-			if err != nil {
-				log.Errorf("Error estimating batch verification to add to eth tx manager: %v", err)
-				a.handleFailureToAddVerifyBatchToBeMonitored(ctx, proof)
-				continue
-			}
-			monitoredTxID := buildMonitoredTxID(proof.BatchNumber, proof.BatchNumberFinal)
-			err = a.EthTxManager.Add(ctx, ethTxManagerOwner, monitoredTxID, sender, to, nil, data, a.cfg.GasOffset, nil)
-			if err != nil {
-				mTxLogger := ethtxmanager.CreateLogger(ethTxManagerOwner, monitoredTxID, sender, to)
-				mTxLogger.Errorf("Error to add batch verification tx to eth tx manager: %v", err)
-				a.handleFailureToAddVerifyBatchToBeMonitored(ctx, proof)
-				continue
+			switch a.cfg.SettlementBackend {
+			case Beethoven:
+				if success := a.settleWithBeethoven(ctx, proof, inputs); !success {
+					continue
+				}
+			default:
+				if success := a.settleDirect(ctx, proof, inputs); !success {
+					continue
+				}
 			}
```

Comment:
WDYT about abstracting this part into some interface. Let's call it e.g. `ISettler` with the following function:

```go
type ISettler interface {
	Settle(ctx context.Context, proof *state.Proof, inputs types.FinalProofInputs) bool
}
```
You would have two implementations e.g. `L1Settler` and `BeethovenSettler` and instantiate the corresponding one based on the `SettlementBackend` "enumeration".

Reply:
Yep, totally agree. This is the correct way to implement that. On it.

Reply:
I've looked closer at the aggregator code and I think we can go with that switch/case cause this will otherwise end up in a fully refactoring of it. Many basic architectural principles are violated here.